streams/src/main/resources/common/message/SubscriptionInfo.json

```diff
@@ -0,0 +1,58 @@
+{
+  "package": "org.apache.kafka.streams.internals.generated",
+  "name": "SubscriptionInfo",
```

Comment:
What I settled on is that the `validVersions` would be a range encompassing all the supported protocol versions. The individual fields would use an open range when they're added (like `3+`), and then use a closed range when they are removed. For example, in KIP-441, we're going to set `validVersions: 1-6` and then set (for example) `prevTasks: 1-5`, while adding new fields `6+`.

This scheme would give us the smallest diff when we do have to bump the protocol version without changing any fields (which happens with some regularity). It also makes the most intuitive sense, IMHO, to use open ranges for the fields when you add them, because you would just assume they'll remain part of the protocol indefinitely.

Reply:
I'd suggest we double check on the semantics of `default values` as we have been bitten while working on KIP-345 before. @abbccdda could you chime in here and double check on the default value / / versions fields and see if they make sense?